deft.el

```diff
@@ -1705,25 +1705,28 @@ Otherwise, quick create a new file."
 (declare-function org-add-link-type "org")
 (declare-function org-open-file-with-emacs "org")
 
+(defun deft-id-at-point ()
+  (substring (expand-file-name (deft-filename-at-point))
+             (1+ (length (expand-file-name deft-directory))))
+  )
 (defun org-deft-store-link ()
   "Store the Deft widget at point as an org-mode link."
   (when (equal major-mode 'deft-mode)
-    (let ((link (concat "deft:" (substring (deft-filename-at-point)
-                                           (1+ (length deft-directory)))))
+    (let ((link (concat "deft:" (deft-id-at-point)))
           (title (deft-file-title (deft-filename-at-point))))
       (org-store-link-props
        :type "deft"
        :link link
        :description title))))
 
 (with-eval-after-load 'org
-  (org-add-link-type
```

Comment:
If possible, can we keep the old code around for backwards compatibility. In other words, can this can be a conditional like so?

```emacs-lisp
(if (fboundp 'org-link-set-parameters)
    (org-link-set-parameters …)
  (org-add-link-type …))
```